torch/_subclasses/meta_utils.py

```diff
@@ -961,11 +1018,24 @@ def tensor_visitor_fn(
                     sizes, strides, _storage_offset = sym_sizes_strides_storage_offset(
                         t, source
                     )
+                    # TODO: This doesn't seem right, where's the MKLDNN'ness
+                    # lol
                     r = callback(
                         lambda: torch.empty_strided(
                             sizes, strides, dtype=t.dtype, device="meta"
                         )
                     )
+                    if self.copy_data:
+                        with torch.no_grad(), no_dispatch():
+                            assert t.size is not None
+                            assert t.stride is not None
+                            r.real_tensor = torch.empty_strided(
```

Comment:
Why is this one built with empty strided and not clone?
Also I would expect to see storage_offset everywhere we have stride?

Reply:
Whenever the tensor has strides, I have to use empty_strided and then copy to restore striding.

The storage offset is an interesting problem. When I have multiple tensors aliasing to the same tensor, I will have done the storage, and then set_ the storage into the tensors. In this case, storage_offset will be set. But there are a bunch of legacy sites which are not first constructing a storage before constructing the tensor. I didn't fix those, but when this happens we can never setup aliasing. That also means that technically the value of storage offset is un-observable. It can be if you do naughty review a tensor outside of its bounds. Mehhhh.

If this ever is a problem, we just need to refactor all the paths to consistently do storage.

Reply:
ok!